keps/sig-node/4622-topologymanager-max-allowable-numa-nodes/README.md

```diff
@@ -0,0 +1,753 @@
+The current hard-coded value of 8 was added as a stop-gap 4 years ago, a configurable option policy option can improve the topology manager behavior and support modern high-end CPUs which are more and more likely to expose more than 8 NUMA cells.
+
+## Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+- Support high-end CPUs with more than 8 NUMA cells.
+
+## Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- This proposal does not aim to modify the existing TopologyManager Policies. It focuses solely on introducing a new policy option to let users configure the maximum supported number of NUMA nodes.
+- It does not address other resource allocation or management aspects within Kubernetes.
+- Improve the topology manager to remove the state explosion.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+We propose to add a new `TopologyManager` policy option called `max-allowable-numa-nodes` to the static TopologyManager policy. It can configure the value of maxAllowableNUMANodes in the TopologyManager. the current hard-coded value of 8 was added as a stop-gap 4 years ago, a configurable option policy option can improve the topology manager to remove the state explosion that occurs when trying to enumerate the possible NUMA affinities and generating their hints.
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+1. As developer in the AI space, I want to use AI accelerators "super chips" which expose ARM cores with more than 8 NUMA nodes.  
+
+2. As user in the high-performance-computing space, I want to enable the sub-NUMA or NUMA-per-socket option of my high-end x86 CPU, which will bring  
+the count of NUMA nodes to exceed 8.  
+
+3. As administrator of edge nodes, I want to use power-efficient yet massively parallel ARM chips which expose more 
```

Comment:
all of those are conflated in a single story, please split in 3 stories

Reply:
sorry, fixed it.